src/main/java/com/iexec/common/security/CipherUtils.java

```diff
@@ -0,0 +1,384 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.common.security;
+
+import com.iexec.common.utils.FileHelper;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.annotation.Nonnull;
+import javax.crypto.Cipher;
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.security.*;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Objects;
+import java.util.Optional;
+
+@Slf4j
+public class CipherUtils {
+
+    // ###############
+    // #     AES     #
+    // ###############
+
+    /**
+     * Generate a 256 bits AES key.
+     * 
+     * @return 256 bits key if success,
+     * empty byte array otherwise.
+     */
+    public static byte[] generateAesKey() {
+        return generateAesKey(256);
+    }
+
+    /**
+     * Generate an AES symmetric key with the
+     * given size.
+     * 
+     * @param size
+     * @return Generated AES key if success,
+     * empty byte array otherwise.
+     */
+    public static byte[] generateAesKey(int size) {
+        try {
+            KeyGenerator generator = KeyGenerator.getInstance("AES");
+            generator.init(size); // The AES key size in number of bits
+            SecretKey secKey = generator.generateKey();
+            return secKey.getEncoded();
+        } catch (NoSuchAlgorithmException e) {
+            log.error("Failed to generate AES key", e);
+            return new byte[0];
+        }
+    }
+
+    /**
+     * Generate a 16 bytes initialization vector.
+     * @return generated binary IV.
+     */
+    public static byte[] generateIv() {
+        return generateIv(16);
+    }
+
+    /**
+     * Generate an initialization vector with
+     * the given size.
+     * @param size
+     * @return generated binary IV.
+     */
+    public static byte[] generateIv(int size) {
+        byte[] iv = new byte[size];
+        new SecureRandom().nextBytes(iv);
+        return iv;
+    }
+
+    /**
+     * Extract IV (initialization vector) from the
+     * encrypted data. The IV occupies the first
+     * 16 bytes of the data.
+     * 
+     * @param dataWithIv
+     * @return Extracted 16 bytes IV
+     */
+    public static byte[] getIvFromEncryptedData(byte[] dataWithIv) {
+        return Arrays.copyOfRange(dataWithIv, 0, 16); // 0 -> 15;
+    }
+
+    /**
+     * Remove prepended IV (initialization vector)
+     * from the encrypted data.
+     * 
+     * @param dataWithIv
+     * @return a new array without the IV.
+     */
+    public static byte[] stripIvFromEncryptedData(byte[] dataWithIv) {
```

Comment:
Maybe soemething like
`getEncryptedDataWithoutIv(..)`
since `stripIvFromEncryptedData(..)` lets think original array will be updated

Reply:
I don't think `strip` does necessarily mean that we are modifying the array itself. `String#strip()` for example does not do that. I can still rename it if you still think it can be confusing.